lib/puppet/type/puppet_authorization_hocon_rule.rb

```diff
@@ -1,65 +1,58 @@
 
+Puppet::Type.newtype(:puppet_authorization_hocon_rule) do
   ensurable do
     defaultvalues
     defaultto :present
   end
 
-  newparam(:name, :namevar => true) do
+  newparam(:name, namevar: true) do
     desc 'An arbitrary name used as the identity of the resource.'
   end
 
   newparam(:path) do
     desc 'The file Puppet will ensure contains the specified setting.'
     validate do |value|
-      unless (Puppet.features.posix? and value =~ /^\//) or (Puppet.features.microsoft_windows? and (value =~ /^.:\// or value =~ /^\/\/[^\/]+\/[^\/]+/))
-        raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'")
-      end
+      raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'") unless (Puppet.features.posix? && value =~ (%r{^/})) || (Puppet.features.microsoft_windows? && (value =~ (%r{^.:/}) || value =~ (%r{^//[^/]+/[^/]+})))
     end
   end
 
-  newproperty(:value, :array_matching => :all) do
+  newproperty(:value, array_matching: :all) do
     desc 'The value of the setting to be defined.'
 
     validate do |val|
-      unless val.is_a?(Hash)
-        raise "Value must be a hash but was #{value.class}"
-      end
+      raise "Value must be a hash but was #{value.class}" unless val.is_a?(Hash)
+
       validate_acl(val)
     end
 
     def validate_acl(val)
-      ["allow", "deny"].each do |rule|
-        if val.has_key?(rule)
-          if val[rule].is_a?(Hash)
-            validate_acl_hash(val[rule], rule)
-          elsif val[rule].is_a?(Array)
-            hashes = val[rule].select {|cur_rule| cur_rule.is_a?(Hash) }
-            hashes.each {|cur_rule| validate_acl_hash(cur_rule, rule) }
-          end
+      %w[allow deny].each do |rule|
+        next unless val.key?(rule)
+
+        case val[rule]
+        when Hash
+          validate_acl_hash(val[rule], rule)
+        when Array
+          hashes = val[rule].select { |cur_rule| cur_rule.is_a?(Hash) }
+          hashes.each { |cur_rule| validate_acl_hash(cur_rule, rule) }
         end
       end
     end
 
     def validate_acl_hash(val, rule)
-      allowed_keys = ["certname", "extensions"]
+      allowed_keys = %w[certname extensions]
       unknown_keys = val.reject { |k, _| allowed_keys.include?(k) }
-      unless unknown_keys.empty?
-        raise "Only one of 'certname' and 'extensions' are allowed keys in a #{rule} hash. Found '#{unknown_keys.keys.join(', ')}'."
-      end
-      unless val.length == 1
-        raise "Only one of 'certname' and 'extensions' are allowed keys in a #{rule} hash."
-      end
+      raise "Only one of 'certname' and 'extensions' are allowed keys in a #{rule} hash. Found '#{unknown_keys.keys.join(', ')}'." unless unknown_keys.empty?
+      raise "Only one of 'certname' and 'extensions' are allowed keys in a #{rule} hash." unless val.length == 1
     end
 
-    def insync?(is)
+    def insync?(_is)
       # make sure all passed values are in the file
       Array(@resource[:value]).each do |v|
```

Comment:
Isn't this whole method equal to:
```suggestion
      values = provider.value.flatten
      Array(@resource[:value]).all? { |v| values.include?(v) }
```